Executes the specified command on a service not working

When you want to execute a command on a service using the Rest API, you got this error:

```
curl -X POST "http://127.0.0.1:2812/api/v3/service/Lobby-1/command"  -H 'accept: application/problem+json' -H 'content-type: application/json' -H 'authorization: Bearer MY_TOKEN'  -d '{"command":"test"}'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   136  100    26  56291  10761 --:--:-- --:--:-- --:--:-- 81000
{"detail":"Unable to decode provided request body.","title":"Exception during decoding","status":400,"type":"exception-during-decoding"}
```

From the documentation too:
<img width="1421" height="468" alt="Image" src="https://github.com/user-attachments/assets/84105b72-5115-46b3-a0ce-873058253fa0" />

Here is debug logs from the module:
```
[04.09 22:22:07.150] DEBUG: Exception while posting request to handler. Details: There was a problem processing annotation @RequestTypedBody: HandlerMethod=[class eu.cloudnetservice.ext.modules.rest.v3.V3HttpHandlerService.handleServiceCommandRequest((String,Map):IntoResponse)]; Parameter=[interface java.util.Map arg1]; DetailDescription=Calling the deserializer of type class eu.cloudnetservice.ext.modules.rest.codec.CloudNetDocumentJsonCodec with the request body raised an exception.
eu.cloudnetservice.driver.document.DocumentParseException: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unexpected value at line 1 column 2 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
    at eu.cloudnetservice.driver.impl.document.gson.GsonDocumentFactory.parse(GsonDocumentFactory.java:112)
    at eu.cloudnetservice.ext.modules.rest.codec.CloudNetDocumentJsonCodec.deserialize(CloudNetDocumentJsonCodec.java:47)
    at eu.cloudnetservice.ext.rest.api.annotation.parser.processor.RequestTypedBodyProcessor.lambda$buildPreprocessor$1(RequestTypedBodyProcessor.java:77)
    at eu.cloudnetservice.ext.rest.api.annotation.parser.ParameterInvocationHint.resolveValue(ParameterInvocationHint.java:49)
    at eu.cloudnetservice.ext.rest.api.annotation.invoke.DefaultHttpHandlerMethodParamResolver.resolveMethodParameter(DefaultHttpHandlerMethodParamResolver.java:84)
    at eu.cloudnetservice.ext.rest.api.annotation.invoke.HttpHandlerMethodContext.handle(HttpHandlerMethodContext.java:64)
    at eu.cloudnetservice.ext.rest.netty.NettyHttpServerHandler.postRequestToHandler(NettyHttpServerHandler.java:306)
    at eu.cloudnetservice.ext.rest.netty.NettyHttpServerHandler.handleMessage(NettyHttpServerHandler.java:233)
    at eu.cloudnetservice.ext.rest.netty.NettyHttpServerHandler.lambda$messageReceived$0(NettyHttpServerHandler.java:143)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
    at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unexpected value at line 1 column 2 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
    at com.google.gson.internal.Streams.parse(Streams.java:58)
    at com.google.gson.JsonParser.parseReader(JsonParser.java:144)
    at com.google.gson.JsonParser.parseReader(JsonParser.java:110)
    at eu.cloudnetservice.driver.impl.document.gson.GsonDocumentFactory.parse(GsonDocumentFactory.java:109)
    ... 13 common frames omitted
Caused by: com.google.gson.stream.MalformedJsonException: Unexpected value at line 1 column 2 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
    at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1817)
    at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:696)
    at com.google.gson.stream.JsonReader.peek(JsonReader.java:547)
    at com.google.gson.internal.Streams.parse(Streams.java:44)
    ... 16 common frames omitted
```

--- a/driver/impl/src/main/java/eu/cloudnetservice/driver/impl/network/netty/memory/MemorySegmentBuffer.java
+++ b/driver/impl/src/main/java/eu/cloudnetservice/driver/impl/network/netty/memory/MemorySegmentBuffer.java
@@ -300,7 +300,8 @@ final class MemorySegmentBuffer
 
     if (this.hasReadableArray()) {
       var srcArray = this.readableArray();
-      System.arraycopy(srcArray, srcPos, dest, destPos, length);
+      var srcArrayOff = Math.toIntExact(this.readSegment.address());
+      System.arraycopy(srcArray, srcArrayOff + srcPos, dest, destPos, length);
     } else {
       var destSegment = MemorySegment.ofArray(dest);
       MemorySegment.copy(this.readSegment, srcPos, destSegment, destPos, length);
@@ -320,9 +321,10 @@ final class MemorySegmentBuffer
 
     if (dest.hasArray() && this.hasReadableArray()) {
       var srcArray = this.readableArray();
+      var srcArrayOff = Math.toIntExact(this.readSegment.address());
       var destArray = dest.array();
-      var destFrom = dest.arrayOffset() + destPos;
-      System.arraycopy(srcArray, srcPos, destArray, destFrom, length);
+      var destArrayOff = dest.arrayOffset();
+      System.arraycopy(srcArray, srcArrayOff + srcPos, destArray, destArrayOff + destPos, length);
     } else {
       var cleanedBuffer = dest.duplicate().clear();
       var destSegment = MemorySegment.ofBuffer(cleanedBuffer);

--- a/driver/impl/src/test/java/eu/cloudnetservice/driver/impl/network/netty/memory/MemorySegmentBufferTest.java
+++ b/driver/impl/src/test/java/eu/cloudnetservice/driver/impl/network/netty/memory/MemorySegmentBufferTest.java
@@ -767,6 +767,29 @@ public class MemorySegmentBufferTest {
 
   @ParameterizedTest
   @MethodSource("bufferTypes")
+  void testCopyWithInternalOffset(Buffer buffer) {
+    buffer.implicitCapacityLimit(18); // only needing 18 bytes at max
+
+    buffer.writeShort((short) 8943);
+    buffer.writeLong(987654321123456789L);
+    buffer.writeLong(-2784823473287483223L);
+    Assertions.assertEquals(0, buffer.readerOffset());
+    Assertions.assertEquals(18, buffer.writerOffset());
+
+    buffer.split(Short.BYTES); // move MemSeg offset to 2
+    Assertions.assertEquals(0, buffer.readerOffset());
+
+    var longTarget = new byte[8];
+    buffer.copyInto(8, longTarget, 0, 8);
+    Assertions.assertEquals(-2784823473287483223L, BYTE_ARRAY_AS_LONG.get(longTarget, 0));
+
+    var longTargetBB = ByteBuffer.allocate(8);
+    buffer.copyInto(8, longTargetBB, 0, 8);
+    Assertions.assertEquals(-2784823473287483223L, longTargetBB.getLong());
+  }
+
+  @ParameterizedTest
+  @MethodSource("bufferTypes")
   void testTransferToWritableByteChannel(Buffer buffer) throws IOException {
     buffer.implicitCapacityLimit(18); // only needing 18 bytes at max
 

